libr/core/cmd_open.c

```diff
@@ -1416,9 +1416,18 @@ static int cmd_open(void *data, const char *input) {
 				bits = r_num_math (core->num, r_str_word_get0 (ptr, 1));
 				arch = r_str_word_get0 (ptr, 0);
 				r_core_bin_set_arch_bits (core, filename, arch, bits);
-				RBinFile *file = r_bin_file_find_by_name (core->bin, filename);
-				if (!file) {
-					eprintf ("Cannot find file %s\n", filename);
+				RBinFile *file;
```

Comment:
initialize to NULL; so you can move the if (!file) check once after trying to open the file